src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -871,32 +882,36 @@ def test_declarations_with_initialisations(fortran_reader):
         """)
 
     inner_st = psyir.walk(Routine)[0].symbol_table
+    asym = inner_st.lookup('a')
+    aasym = inner_st.lookup('aa')
+    bsym = inner_st.lookup('b')
+    csym = inner_st.lookup('c')
+    dsym = inner_st.lookup('d')
+    esym = inner_st.lookup('e')
+    fsym = inner_st.lookup('f')
     # All initialisation variables are DataSymbols
-    assert isinstance(inner_st.lookup('a'), DataSymbol)
-    assert isinstance(inner_st.lookup('b'), DataSymbol)
-    assert isinstance(inner_st.lookup('c'), DataSymbol)
-    assert isinstance(inner_st.lookup('d'), DataSymbol)
-    assert isinstance(inner_st.lookup('e'), DataSymbol)
-    assert isinstance(inner_st.lookup('f'), DataSymbol)
-
-    # When it is not a parameter they are unknown interface and datatype
-    assert isinstance(inner_st.lookup('a').interface, UnknownInterface)
-    assert isinstance(inner_st.lookup('b').interface, UnknownInterface)
-    assert isinstance(inner_st.lookup('d').interface, UnknownInterface)
-    assert isinstance(inner_st.lookup('e').interface, UnknownInterface)
-    assert isinstance(inner_st.lookup('a').datatype, UnknownFortranType)
-    assert isinstance(inner_st.lookup('b').datatype, UnknownFortranType)
-    assert isinstance(inner_st.lookup('d').datatype, UnknownFortranType)
-    assert isinstance(inner_st.lookup('e').datatype, UnknownFortranType)
-
-    # When it is a parameter the interface, type and constant_value is defined
-    assert isinstance(inner_st.lookup('c').interface, DefaultModuleInterface)
-    assert isinstance(inner_st.lookup('c').datatype, ScalarType)
-    assert isinstance(inner_st.lookup('c').constant_value, Literal)
-    assert isinstance(inner_st.lookup('f').interface, AutomaticInterface)
-    assert isinstance(inner_st.lookup('f').datatype, ScalarType)
-    assert isinstance(inner_st.lookup('f').constant_value, Literal)
-    assert isinstance(inner_st.lookup('f').interface, AutomaticInterface)
+    assert all(isinstance(sym, DataSymbol) for sym in
+               [asym, aasym, bsym, csym, dsym, esym, fsym])
+
+    # All of the module symbols should have a DefaultModuleInterface.
+    assert all(isinstance(sym.interface, DefaultModuleInterface) for sym in
+               [asym, aasym, csym])
```

Comment:
Updated. I've changed the DataSymbol class so that `is_constant` implies a StaticInterface (if no interface is specified).
I need to think about CommonBlockInterface.

Reply:
![image](https://github.com/stfc/PSyclone/assets/17409212/4329ffa8-0a05-45ba-abd9-361a2c2cb5fc)

Reply:
Fortran2008 has:
![image](https://github.com/stfc/PSyclone/assets/17409212/f6f2a21f-a564-4a23-a85f-7209293bb9b7)

Reply:
gfortran also won't let me declare a parameter as a member of a common block.

Reply:
Since a parameter must, by definition, have an initial value. I've extended the fparser2 frontend to reject symbols that appear in common blocks but that have initial values.